trin-core/src/portalnet/overlay_service.rs

```diff
@@ -916,17 +926,49 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
                 "[{:?}] Skipping processing for content connection ID {}",
                 self.protocol, id
             ),
-            Content::Content(_) => {
-                debug!(
-                    "[{:?}] Skipping processing for content bytes",
-                    self.protocol
-                )
-            }
+            Content::Content(content) => self.process_received_content(content, request).await,
             Content::Enrs(enrs) => self
                 .process_discovered_enrs(enrs.into_iter().map(|ssz_enr| ssz_enr.into()).collect()),
         }
     }
 
+    async fn process_received_content(&mut self, content: ByteList, request: Request) {
+        let find_content = match request {
+            Request::FindContent(find_content) => find_content,
+            _ => {
+                error!("Unable to process received content: Invalid request message.");
+                return;
+            }
+        };
+        let content_key = match TContentKey::try_from(find_content.content_key) {
+            Ok(val) => val,
+            Err(_) => {
+                error!("Unable to process received content: Invalid content key.");
+                return;
+            }
+        };
+        let ck_clone = content_key.clone();
+        let content_clone = content.clone();
+        self.validator
+            .validate_content(ck_clone, content_clone)
+            .await;
```

Comment:
So here is where I expect an `Err` check and then an early return before storage if it was invalid.